classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -269,13 +499,14 @@ function updateObject($object) {
 
 	/**
 	 * Identify the dao name and update method for the given object.
-	 * @param $object Issue|PublishedArticle
+	 * @param $object Issue|PublishedArticle|ArticleGalley
 	 * @return array (DAO, Method)
 	 */
 	function getDaoMethod($object) {
 		$configurations = array(
```

Comment:
Just thinking structurally that we can make some improvements as we standardize the function naming etc. For example, in the DataObject class, we could add

    function getDAO() {
        assert(false);
    }

...then start to implement this in specific subclasses. Then this function could be entirely removed, and the `DOIPubIdExportPlugin::updateObject` function could call `getDAO` on the object and then invoke `updateObject` on that directly.